packages/delivery-react-native/make-safe.js

```diff
@@ -0,0 +1,79 @@
+const isArray = require('@bugsnag/core/lib/es-utils/is-array')
+
+const isSafeLiteral = (obj) => (
+  typeof obj === 'string' || obj instanceof String ||
+  typeof obj === 'number' || obj instanceof Number ||
+  typeof obj === 'boolean' || obj instanceof Boolean
+)
+
+const isError = o => (
+  o instanceof Error || /^\[object (Error|(Dom)?Exception)]$/.test(Object.prototype.toString.call(o))
+)
+
+const throwsMessage = err => '[Throws: ' + (err ? err.message : '?') + ']'
+
+const safelyGetProp = (obj, propName) => {
+  try {
+    return obj[propName]
+  } catch (err) {
+    return throwsMessage(err)
+  }
+}
+
+/**
+ * Similar to `safe-json-stringify` this function rebuilds an object graph to be safe for the ReactNative JS Bridge.
+ * This requirement is different to `JSON.parse(safeJsonStringify(data))` in three key ways:
+ * - `toJSON` methods are not called
+ * - we leave more types intact for the JS Bridge to handle
+ * - there is no redaction or fixed depth limit
+ *
+ * @param data the value to be made safe for the ReactNative bridge
+ * @returns a safe version of the given `data`
+ */
+module.exports = function (data) {
+  const seen = []
+
+  const visit = (obj) => {
+    if (obj === null || obj === undefined) return obj
+
+    if (isSafeLiteral(obj)) {
+      return obj
+    }
+
+    if (isError(obj)) {
+      return visit({ name: obj.name, message: obj.message })
+    }
+
+    if (seen.includes(obj)) {
+      // circular references are replaced and marked
+      return '[Circular]'
+    }
+
+    // handle arrays, and all iterable non-array types (such as Set)
+    if (isArray(obj) || obj[Symbol.iterator]) {
```

Comment:
Do all versions of RN support `Symbol`?

I can't really find anything very helpful, but there's [this error from not having a `Symbol` global on SO](https://stackoverflow.com/questions/50504532/react-native-cant-find-variable-symbol) which may mean we need to be careful about using `Symbol`

Something like this might be safer:

```suggestion
    if (isArray(obj) || (Symbol && obj[Symbol.iterator])) {
```

Reply:
`Symbol` is available on all the RN versions we support, but I'm happy to addd the secondary check for safety sake.

Reply:
If it's supported everywhere then it's probably fine as-is, I just couldn't find anything definitive in a quick search